### PR TITLE
[RFR] Add Restful.allUrl() and Restful.oneUrl() methods to allow custom urls

### PR DESCRIPTION
**Pull request by jeromemacias**

### CODE REVIEW

File: src/model/member.js [hunk 1]
```diff
@@ -5,13 +5,26 @@ import responseBuilder from 'service/responseBuilder';
...
 export default function member(name, id, parent) {
-    var refEndpoint = endpoint([parent.url(), name].join('/'), parent());
+    var url = parent.customUrl && parent.customUrl() ? parent.customUrl() : [parent.url(), name, id].join('/');
+
+    var refEndpoint = endpoint(url, parent());
 
     var model = {
+        _url: null,
+
+        customUrl(url) {
```
Comment (jpetitcolas) [inline src/model/member.js]: Shouldn't we factorize `_url` and `customUrl` into `Resource` model?

File: src/model/member.js [hunk 2]
```diff
@@ -51,12 +64,24 @@ export default function member(name, id, parent) {
             return member(name, id, model);
         },
 
+        oneUrl(name, url) {
```
Comment (jpetitcolas) [inline src/model/member.js]: Same

File: src/model/member.js [hunk 3]
```diff
@@ -51,12 +64,24 @@ export default function member(name, id, parent) {
...
+            this.customUrl(url);
+
+            return this.one(name, null);
+        },
+
         all(name) {
             return collection(name, model);
         },
 
+        allUrl(name, url) {
```
Comment (jpetitcolas) [inline src/model/member.js]: Same

**Comment by jpetitcolas** [comment]
:+1:

File: src/model/member.js (hunk 1 above)
Comment (jeromemacias) [inline src/model/member.js]: Collection extend resource but don't have `one()`and `all()` method, so we can't factorize (cc @RobinBressan).

**Pull request merged**